packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -1060,11 +1058,13 @@ class PopupMenuButton<T> extends StatefulWidget {
   /// and the button will behave like an [IconButton].
   final Widget? icon;
 
-  /// The offset is applied relative to the initial position
-  /// set by the [position].
+  /// The offset returned by this method is used to place the
+  /// menu button. The origin of this offset's coordinate system
+  /// is at the top left of the menu button.
+  /// The size of the menu button will be passed to this method.
   ///
   /// When not set, the offset defaults to [Offset.zero].
-  final Offset offset;
```

Comment:
Removing the offset property here might be risky since it's been created for a long time and might break other APIs. It would be great if there's another way to handle the issue you talked about without deleting offset.

Reply:
@QuncCccccc  maybe making `offsetBuilder` nullable (optional parameter) and add an assert that only one of them can be provided at a time. This way we can keep both offset and `offsetBuilder` and the code will stay backwards compatible. Do you think this is good solution?